File: tests/runtimes/python/python_globs.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { PythonRuntime } from "@typegraph/sdk/runtimes/python";
import { Policy, t, typegraph } from "@typegraph/sdk";

const tpe = t.struct({
  a: t.string(),
  b: t.list(t.either([t.integer(), t.string()])),
});

export const tg = await typegraph("python_globs", (g: any) => {
  const python = new PythonRuntime();
  const pub = Policy.public();

  g.expose({
    testGlob: python
      .import(t.struct({ input: tpe }), tpe, {
        name: "identity",
        module: "py/hello.py",
        deps: ["py/**/*.py"],
      })
      .withPolicy(pub),
  });
});

File: tests/runtimes/python/python_globs.py
# Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
# SPDX-License-Identifier: MPL-2.0

from typegraph import t, typegraph
from typegraph.graph.typegraph import Graph
from typegraph.policy import Policy
from typegraph.runtimes.python import PythonRuntime


@typegraph()
def python_globs(g: Graph):
    public = Policy.public()
    python = PythonRuntime()

    g.expose(
        test_glob=python.import_(
            t.struct({"name": t.string()}),
            t.string(),
            module="py/hello.py",
            deps=["py/nested/*.py"],
            name="sayHello",
        ).with_policy(public),
    )
